rules/recording.go

```diff
@@ -93,6 +93,16 @@ func (rule *RecordingRule) Eval(ctx context.Context, ts time.Time, query QueryFu
 
 		sample.Metric = lb.Labels()
 	}
+
+	// Check that the rule does not produce identical metrics after applying
+	// labels #5529
```

Comment:
I'd put the reference to the issue in the unittest. This isn't some weird corner case, it's a design thing so feels weird to document by bug.